Gems/AtomLyIntegration/AtomViewportDisplayIcons/Code/Source/AtomViewportDisplayIconsSystemComponent.cpp

```diff
@@ -354,20 +361,22 @@ namespace AZ::Render
     {
         // Once the shader is loaded, register it with the dynamic draw context
         Data::Asset<RPI::ShaderAsset> shaderAsset = asset;
-        AtomBridge::PerViewportDynamicDraw::Get()->RegisterDynamicDrawContext(m_drawContextName, [shaderAsset](RPI::Ptr<RPI::DynamicDrawContext> drawContext)
+        AtomBridge::PerViewportDynamicDraw::Get()->RegisterDynamicDrawContext(m_drawContextName, [shaderAsset](RPI::Ptr<RPI::DynamicDrawContext> dynamicDraw)
             {
                 AZ_Assert(shaderAsset->IsReady(), "Attempting to register the AtomViewportDisplayIconsSystemComponent"
                     " dynamic draw context before the shader asset is loaded. The shader should be loaded first"
                     " to avoid a blocking asset load and potential deadlock, since the DynamicDrawContext lambda"
                     " will be executed during scene processing and there may be multiple scenes executing in parallel.");
 
                 Data::Instance<RPI::Shader> shader = RPI::Shader::FindOrCreate(shaderAsset);
-                drawContext->InitShader(shader);
-                drawContext->InitVertexFormat(
-                    { {"POSITION", RHI::Format::R32G32B32_FLOAT},
-                     {"COLOR", RHI::Format::R8G8B8A8_UNORM},
-                     {"TEXCOORD", RHI::Format::R32G32_FLOAT} });
-                drawContext->EndInit();
+                dynamicDraw->InitShader(shader);
+                dynamicDraw->InitVertexFormat({ { "POSITION", RHI::Format::R32G32B32_FLOAT },
+                                                { "COLOR", RHI::Format::R8G8B8A8_UNORM },
+                                                { "TEXCOORD", RHI::Format::R32G32_FLOAT } });
+                dynamicDraw->AddDrawStateOptions(
+                    AZ::RPI::DynamicDrawContext::DrawStateOptions::PrimitiveType |
+                    AZ::RPI::DynamicDrawContext::DrawStateOptions::BlendMode | AZ::RPI::DynamicDrawContext::DrawStateOptions::DepthState);
```

Comment:
question to Atom: This might be redundant or overkill, let me know if it's worth removing but this is present in LyShine equivalent code (it might be worth keeping it but updating the flags slightly)

Reply:
This isn't necessary unless you need to change the blend mode to value difference than the blend mode defined in the .shader file, which doesn't look like the case. 

I think you can removed it.

Reply:
Okay perfect I'll get rid of this :+1: